docs/core_api.md

```diff
@@ -0,0 +1,61 @@
+# Core API
+
+The core API of ErisCasper.java is designed to be minimal and powerful.
+
+It exposes the concepts of `Bot`s, the event stream (`Observable<Event>`), `Action`s, and `Repository`s as
+described below.
+
+There also exists the concept `Gateway` and `Routes`.
+These are lower level wrappers around the Discord API and considered internal
+to ErisCasper.java.
+This means their use is heavily discouraged, but they are available to bot
+developers if required.
+
+
+## Bot
+
+A `Bot` is a `Function` run with access to a `BotContext` and produces a `Completable`.
+
+The event stream, `Repository`s, and ability to create `Action`s are 
+accessible via the `BotContext`.
+
+The return type is a `Completable` as defined by RxJava.
+This Completable will be subscribed to by ErisCasper.java.
+
+`Bot`s can be composed. Multiple implementations of `Bot` may be merged into a single `Bot`.
+
+
+## Event Stream
+
+The event stream is the events sent from Discord.
```

Comment:
`consists of the events sent from discord`
A stream would consist of items, not be the array of events. imo it works better. Up for debate.